### PR TITLE
fix(client): editing SwaggerUI with auth-basic

### PR DESCRIPTION
**Pull request by voytovichdm**
Close: #[8058](https://github.com/amplication/amplication/issues/8058)

## PR Details

Added the necessary authentication type to the ServiceSettings.authProvider field when modifying authentication plugins.

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-client/src/Plugins/PluginsCatalog.tsx
Comment (overbit) [inline packages/amplication-client/src/Plugins/PluginsCatalog.tsx:1]: Thanks @voytovichdm for the PR. Unfortunately we cannot merge this as it goes against the amplication design. 
### TL;DR; the change must live within the `auth-basic` plugin instead of `amplication-client`


With plugins, `data-service-generator` should not be aware of which plugins are installed and each plugin will perform the necessary operations to produce the expected generated code.

In the old days, client, server and dsg where more aware of things like authentication strategy, today fully managed by this plugins. Of course, we would need to clean up all code that is currently still rely on this  `authProvider` property and I'm opening an issue for this.

**Review by overbit — CHANGES_REQUESTED** [review]
